micro_sam/sam_annotator/annotator_2d.py

```diff
@@ -191,6 +217,41 @@ def annotator_2d(
     v: Optional[Viewer] = None,
     predictor: Optional[SamPredictor] = None,
     precompute_amg_state: bool = False,
+) -> Optional[Viewer]:
+    """This function can be called in python (or from CLI) to start the annotation tool"""
+    if v is None:
+        v = napari.Viewer()
+
+    v.add_image(raw)
+    v = annotator_2d_plugin(
+        raw,
+        embedding_path=embedding_path,
+        show_embeddings=show_embeddings,
+        segmentation_result=segmentation_result,
+        precompute_amg_state=precompute_amg_state,
+        tile_shape=tile_shape,
+        halo=halo,
+        predictor=predictor,
+    )
+
+    if return_viewer is True:
+        return v
+
+    napari.run()
+
+
+@magic_factory(call_button="Start 2D annotation")
+def annotator_2d_plugin(
+    raw: ImageData,
+    v: napari.Viewer,
+    embedding_path: os.PathLike = "./embeddings",
+    show_embeddings: bool = False,
+    segmentation_result: Optional[LabelsData] = None,
+    precompute_amg_state: bool = False,
+    model_type: AVAILABLE_MODELS = AVAILABLE_MODELS[_DEFAULT_MODEL],
+    tile_shape: Tuple[int, int] = (None, None),
+    halo: Tuple[int, int] = (None, None),
+    predictor: Optional[SamPredictor] = None,  # not accessible via magicgui widget, not a recognised type
```

Comment:
Regarding the comment: that's ok. This doesn't make sense to pass via napari anyways. The predictor is currently exposed for 2 reasons:
1. To pass the already initialized predictor in the `image_series_annotator`, where the logic for image embeddings is done externally.
2. To enable passing predictors that were loaded from a custom checkpoint.

As far as I understand your comment, it doesn't hurt to leave it in for now. **After we have merged this PR** I can rethink a bit how this is handled because: re 1. we need to change the `image_series_annotator` logic anyways and re 2.: it would be good to enable loading of custom models also from the napari plugin. (Here: custom model means that we are using a url or filepath to specifiy the checkpoint rather than one of the names from `AVAILABLE_MODELS`).